*Author : Pavithra Rajesh*

# **WEEK ONE**

 **Foundations + Data Understanding**

----------------------------------------------------------------------------------------------------------------------------------------------------------------

📚 Dataset Selection & Justification

Dataset used : **E-Commerce Sales Analysis**

Source : *[Kaggle](https://www.kaggle.com/datasets/fahmidachowdhury/e-commerce-sales-analysis)*

1. Why this domain?

 E‑commerce deals with real world data. Because companies really rely on data to make smart business decisions. Here, I am trying to find which products sell the most,about the pricing, and customer engagement etc. 

2. What problem can be solved using this data?

Im trying to understand e-commerce sales performance by identifying top-selling products, impact of pricing and discounts, helping businesses make data-driven decisions instead of assumptions.this includes trying to answer questions like 
- What is selling well and what is not?
- Which products or categories bring real money to the business?
- When do people buy more — which days or months matter most? etc

📚DATASET OVERVIEW

This dataset contains e-commerce sales data for multiple products across different categories, capturing pricing details, customer feedback, and monthly sales performance over a one-year period. Each record represents a product along with its price, review score, total number of reviews, and sales figures for each month.Dataset contains 1000 rows and 18 columns.The columns are 
- product_id
- product_name
- category
- price
- review_score
- review_count
- sales_month_1
- sales_month_2
- sales_month_3
- sales_month_4
- sales_month_5
- sales_month_6
- sales_month_7
- sales_month_8
- sales_month_9
- sales_month_10
- sales_month_11
- sales_month_12







 📚Python Fundamentals Applied

1. Load dataset

In [1]:
import pandas as pd

In [2]:
df=pd.read_csv('ecommerce_sales_analysis.csv') #to load the dataset

In [3]:
df.head(5) #to view the first 5 rows of the dataset

product_id product_name        category   price  review_score  \
0           1    Product_1        Clothing  190.40           1.7   
1           2    Product_2  Home & Kitchen  475.60           3.2   
2           3    Product_3            Toys  367.34           4.5   
3           4    Product_4            Toys  301.34           3.9   
4           5    Product_5           Books   82.23           4.2   

   review_count  sales_month_1  sales_month_2  sales_month_3  sales_month_4  \
0           220            479            449             92            784   
1           903             21            989            861            863   
2           163            348            558            567            143   
3           951            725            678             59             15   
4           220            682            451            649            301   

   sales_month_5  sales_month_6  sales_month_7  sales_month_8  sales_month_9  \
0            604            904            446            603            807   
1            524            128            610            436            176   
2            771            409            290            828            340   
3            937            421            670            933             56   
4            620            293            411            258            854   

   sales_month_10  sales_month_11  sales_month_12  
0             252             695             306  
1             294             772             353  
2             667             267             392  
3             157             168             203  
4             548             770             257

2.  Print dataset info

In [4]:
df.info() #to get a summary of the dataset

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 18 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   product_id      1000 non-null   int64  
 1   product_name    1000 non-null   object 
 2   category        1000 non-null   object 
 3   price           1000 non-null   float64
 4   review_score    1000 non-null   float64
 5   review_count    1000 non-null   int64  
 6   sales_month_1   1000 non-null   int64  
 7   sales_month_2   1000 non-null   int64  
 8   sales_month_3   1000 non-null   int64  
 9   sales_month_4   1000 non-null   int64  
 10  sales_month_5   1000 non-null   int64  
 11  sales_month_6   1000 non-null   int64  
 12  sales_month_7   1000 non-null   int64  
 13  sales_month_8   1000 non-null   int64  
 14  sales_month_9   1000 non-null   int64  
 15  sales_month_10  1000 non-null   int64  
 16  sales_month_11  1000 non-null   int64  
 17  sales_month_12  1000 non-null   in

What we can see from this dataset:
- The dataset has **1000 rows** which starts from index 0 to index 999 and **18 columns**.
- Every column has 1000 non-null values hence there is no missing values here.


3. Use Python basics

In [5]:
top_category = df['category'].value_counts().idxmax()#to find the category with the highest number of products
print(top_category)


Books


*Books is the category with the largest number of products listed on the platform. which means  that the company is focused more on providing  books than other categories.*

In [6]:
cheapest_product=df.loc[df.price.idxmin(),"product_name"] #to find the cheapest product in the dataset
print(cheapest_product)

Product_822


*The platform offers products at very low prices, with Product_822 =cheapest.*

In [7]:
sales= df[['sales_month_1','sales_month_2','sales_month_3']].sum() #to calculate total sales for each month
print(sales)

sales_month_1    498306
sales_month_2    507661
sales_month_3    506739
dtype: int64


In [8]:

sale={"Month 1":sales['sales_month_1'],"Month 2":sales['sales_month_2'],"Month 3":sales['sales_month_3']}#to store the total sales in a dictionary
print(sale)

{'Month 1': np.int64(498306), 'Month 2': np.int64(507661), 'Month 3': np.int64(506739)}


In [9]:
total = 0

for value in df['sales_month_1']: #to calculate total sales in month 1
    total = total + value

print(total)


498306


In [10]:
cheap_products=df[df.price<100]#to find all products with price less than 100
print(cheap_products)

     product_id product_name        category  price  review_score  \
4             5    Product_5           Books  82.23           4.2   
5             6    Product_6            Toys  82.22           3.6   
6             7    Product_7     Electronics  33.75           3.8   
10           11   Product_11            Toys  15.19           1.9   
14           15   Product_15        Clothing  95.00           3.8   
..          ...          ...             ...    ...           ...   
977         978  Product_978          Sports  30.46           2.7   
982         983  Product_983  Home & Kitchen  57.77           2.8   
983         984  Product_984  Home & Kitchen  42.19           2.9   
995         996  Product_996  Home & Kitchen  50.33           3.6   
997         998  Product_998          Sports  72.73           1.3   

     review_count  sales_month_1  sales_month_2  sales_month_3  sales_month_4  \
4             220            682            451            649            301   
5        

In [11]:
print(df['price'].mean())
#to calculate the average price of all products


247.67713


In [12]:
print(df['sales_month_1'].sum())
#to calculate total sales in month 1

498306


In [13]:
print(df['review_score'].median())


3.1


In [14]:
print(df.describe())
# with this we get statistical summary of numerical columns in the dataset instead of just wrting each separately.

        product_id        price  review_score  review_count  sales_month_1  \
count  1000.000000  1000.000000   1000.000000   1000.000000    1000.000000   
mean    500.500000   247.677130      3.027600    526.506000     498.306000   
std     288.819436   144.607983      1.171243    282.269932     289.941478   
min       1.000000     7.290000      1.000000      1.000000       0.000000   
25%     250.750000   121.810000      2.000000    283.750000     245.500000   
50%     500.500000   250.920000      3.100000    543.000000     507.500000   
75%     750.250000   373.435000      4.000000    772.000000     740.750000   
max    1000.000000   499.860000      5.000000    999.000000    1000.000000   

       sales_month_2  sales_month_3  sales_month_4  sales_month_5  \
count    1000.000000    1000.000000    1000.000000    1000.000000   
mean      507.661000     506.739000     503.823000     487.194000   
std       285.992689     294.010873     286.645567     287.844324   
min         2.000000 

In this data :

In [15]:
print(f'The cheapest product is {cheapest_product}')

The cheapest product is Product_822


In [16]:
print(f'The expensive product is {df.loc[df["price"].idxmax(), "product_name"]}')

The expensive product is Product_532


In [17]:
cheap_price = df.loc[df.price.idxmin(), 'price']
exp_price = df.loc[df.price.idxmax(), 'price']
print(f'Price difference: {exp_price - cheap_price}')


Price difference: 492.57


In [18]:
print(f'The datatypes are {df.dtypes}')


The datatypes are product_id          int64
product_name       object
category           object
price             float64
review_score      float64
review_count        int64
sales_month_1       int64
sales_month_2       int64
sales_month_3       int64
sales_month_4       int64
sales_month_5       int64
sales_month_6       int64
sales_month_7       int64
sales_month_8       int64
sales_month_9       int64
sales_month_10      int64
sales_month_11      int64
sales_month_12      int64
dtype: object


In [19]:
df.isnull().sum()

product_id        0
product_name      0
category          0
price             0
review_score      0
review_count      0
sales_month_1     0
sales_month_2     0
sales_month_3     0
sales_month_4     0
sales_month_5     0
sales_month_6     0
sales_month_7     0
sales_month_8     0
sales_month_9     0
sales_month_10    0
sales_month_11    0
sales_month_12    0
dtype: int64

This shows there isnt any null values in the data.

In [20]:

monthly_sales = df[['sales_month_1','sales_month_2','sales_month_3',
                    'sales_month_4','sales_month_5','sales_month_6',
                    'sales_month_7','sales_month_8','sales_month_9',
                    'sales_month_10','sales_month_11','sales_month_12']].sum()
best_month = monthly_sales.idxmax()
best_sales = monthly_sales.max()

print("Monthly totals:\n", monthly_sales)
print(f"The month with highest sales is {best_month} with {best_sales} total sales")


Monthly totals:
 sales_month_1     498306
sales_month_2     507661
sales_month_3     506739
sales_month_4     503823
sales_month_5     487194
sales_month_6     491653
sales_month_7     507011
sales_month_8     504569
sales_month_9     491934
sales_month_10    514798
sales_month_11    505838
sales_month_12    500386
dtype: int64
The month with highest sales is sales_month_10 with 514798 total sales


In [21]:
print(f'Average price of all products: {df["price"].mean()}')

Average price of all products: 247.67713


In [22]:
print(f'Average review score: {df["review_score"].mean().round(2)}')

Average review score: 3.03
